src/core/config/border_routing.h

```diff
@@ -77,13 +77,13 @@
 #endif
 
 /**
- * @def OPENTHREAD_CONFIG_BORDER_ROUTING_NAT64_ENABLE
+ * @def OPENTHREAD_CONFIG_NAT64_BORDER_ROUTING_ENABLE
  *
  * Define to 1 to enable Border Routing NAT64 support.
  *
  */
-#ifndef OPENTHREAD_CONFIG_BORDER_ROUTING_NAT64_ENABLE
-#define OPENTHREAD_CONFIG_BORDER_ROUTING_NAT64_ENABLE 0
+#ifndef OPENTHREAD_CONFIG_NAT64_BORDER_ROUTING_ENABLE
+#define OPENTHREAD_CONFIG_NAT64_BORDER_ROUTING_ENABLE 0
```

Comment:
How about defining a `config/nat64.h` for all NAT64 relate config and moving this to this header?

- General approach for `OPENTHREAD_CONFIG` is that we try to use the prefix to indicate which module it belongs to. 
- For example, we have `OPENTHREAD_CONFIG_SRP_SERVER_ENABLE` in `config/srp_server.h` and then all other configs related to this start with `OPENTHREAD_CONFIG_SRP_SERVER_{configname}`.
- May make sense to do the same for all `NAT64` related configs as well. 
 
Thoughts?

Reply:
Agree

BTW, I'm thinking `NAT64_BORDER_ROUTING` as an "extension" of `BorderRouter::RoutingManager`, it belongs to NAT64, but it lives in the routing manager, and it supports NAT64 by extending the functions of the routing manager.

Reply:
Yes. Sounds good to me (it can be part of another module but high level a NAT64 related feature)).